airbyte-webapp/src/views/Connector/ConnectorForm/components/Property/SecretConfirmationControl.tsx

```diff
@@ -0,0 +1,114 @@
+import { useField, useFormikContext } from "formik";
+import React, { useEffect, useRef, useState } from "react";
+import { FormattedMessage } from "react-intl";
+import styled from "styled-components";
+
+import { Button } from "components/ui/Button";
+import { Input } from "components/ui/Input";
+import { SecretTextArea } from "components/ui/SecretTextArea";
+
+const ComponentContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+`;
+
+interface SecretConfirmationControlProps {
+  showButtons?: boolean;
+  name: string;
+  multiline: boolean;
+  disabled?: boolean;
+  error?: boolean;
+}
+
+const SecretConfirmationControl: React.FC<SecretConfirmationControlProps> = ({
+  showButtons,
+  disabled,
+  multiline,
+  name,
+  error,
+}) => {
+  const [field, , helpers] = useField(name);
+  const [previousValue, setPreviousValue] = useState<unknown>(undefined);
+  const isEditInProgress = Boolean(previousValue);
+  const controlRef = useRef<HTMLElement>(null);
+
+  const { dirty, touched } = useFormikContext();
+
+  const component =
+    multiline && (isEditInProgress || !showButtons) ? (
+      <SecretTextArea
+        {...field}
+        autoComplete="off"
+        value={field.value ?? ""}
+        rows={3}
+        disabled={disabled}
+        error={error}
+      />
+    ) : (
+      <Input
+        {...field}
+        autoComplete="off"
+        value={field.value ?? ""}
+        type="password"
+        disabled={disabled}
+        error={error}
```

Comment:
Something I noticed while testing: the secret input border is not set to red when it has an error
<img width="788" alt="Screen Shot 2022-12-12 at 11 58 57 AM" src="https://user-images.githubusercontent.com/22731524/207142356-8493846d-0496-479c-ae16-e83e75a274b9.png">

I think this is because the `error` prop is not being passed to this component from `Control.tsx`

Reply:
Good catch, seems like it's like that on master too. Should be fixed:
<img width="642" alt="Screenshot 2022-12-12 at 21 24 38" src="https://user-images.githubusercontent.com/1508364/207147231-081b47d6-aafa-451d-82d1-b199aff46411.png">